twisted/python/_pydoctor.py

```diff
@@ -0,0 +1,116 @@
+
+This module is DEPRECATED. It has been split off into a third party
+package, Twisted %(projectName)s. Please see %(projectURL)s.
+
+This is just a place-holder that imports from the third-party %(projectName)s
+package for backwards compatibility. To use it, you need to install
+that package.
+""" % {'moduleDesc': moduleDesc,
+       'projectName': projectName,
+       'projectURL': projectURL}
+        self.builder.current.docstring = modoc
+
+
+
+class TwistedASTBuilder(zopeinterface.ZopeInterfaceASTBuilder):
+    ModuleVistor = TwistedModuleVisitor
+
+
+
+class TwistedSphinxInventory(SphinxInventory):
+    """
+    Custom SphinxInventory to work around broken external references to
+    Sphinx.
+
+    All exceptions should be reported upstream and a comment should be created
+    with a link to the upstream report.
+    """
+
+    def getLink(self, name):
+        """
+        Return URL for `name` or None if no URL is found.
+        """
+        result = super(TwistedSphinxInventory, self).getLink(name)
+        if result is not None:
+            # We already got a link. Look no further.
+            return result
+
+        if name.startswith('zope.interface.'):
+            # This is a link from zope.interface. which is not advertised in
+            # the Sphinx inventory.
+            # See if the link is a known broken link which should be handled
+            # as an exceptional case.
+            # We get the base URL from IInterface which is assume that is
+            # always and already well defined in the Sphinx index.
+            base_url, _ = self._links.get(
+                'zope.interface.interfaces.IInterface')
+
+            if name  == 'zope.interface.adapter.AdapterRegistry':
+                # FIXME:
+                # https://github.com/zopefoundation/zope.interface/issues/41
+                relative_link = 'adapter.html'
+            else:
+                # Not a known exception.
+                relative_link = None
+
+            if relative_link is None:
+                return None
+
+            return '%s/%s' % (base_url, relative_link)
+
+        return result
+
+
+
+class TwistedSystem(zopeinterface.ZopeInterfaceSystem):
+    """
+    Maybe class used to set up all pydoctor system.
+    """
+    defaultBuilder = TwistedASTBuilder
+
+    def __init__(self, options=None):
+        super(TwistedSystem, self).__init__(options=options)
+        # Use custom SphinxInventory so that we can resolve valid L{} markup
+        # for which the Sphinx inventory is not published or broken.
+        self.intersphinx = TwistedSphinxInventory(
+            logger=self.msg, project_name=self.projectname)
+
+    def privacyClass(self, obj):
+        o = obj
+        if o.fullName() == 'twisted.test':
+            # Match this package exactly, so that proto_helpers
+            # below is visible
+            return model.PrivacyClass.VISIBLE
+        while o:
+            if o.fullName() == 'twisted.words.xish.yappsrt':
```

Comment:
This shouldn't be there. This module is not in Twisted and might be a leftover from a review on the port of xish to Python 3.